### ISSUE TITLE
V3 Beta - errors not reported from preload context

### ISSUE THREAD

**Issue by timfish** [getsentry/sentry-electron#372]
## Versions + Platform

- [x] SDK version - `@sentry/electron@v3.0.0-beta.1`
- [x] Electron version - `all`
- [x] Platform - `all`

## Description

With v3 beta, errors are not reported from the preload context.

**Comment by timfish** [comment]
I'm nowhere near a machine to test this but you can probably work around this by initialising the render code in the preload too:

```
import '@sentry/electron/preload';
Import { init } '@sentry/electron/renderer';

init({});
```

If you're using context isolation, you will need to init the renderer code in the preload AND the isolated renderer context.

**Comment by timfish** [comment]
I think reporting errors from the preload context should work out of the box by default but this adds some annoying complications that vary by the users Electron configuration.

For non-bundled apps, we automatically inject the preload script which is currently well under 1kB in size. If we want to support reporting errors from this context with zero config, we'll need to include all the renderer code which is currently ~78kB. If we include this code by default, we'll also need to call `init` and users will not be able to change the settings passed to `init`. In the preload context, passing options to `init` might not be a huge issue except...

If users have `contextIsolation` disabled (not the default and not recommended, but very common), the preload and renderer context are not distinct so once the SDK has been `init`'ed  in the preload, it's effectively setup in the renderer and there is no way to pass option to `init`. 

Why not let users just create their own preload script and choose if they want to support error reporting from this context and pass custom `init` options? Well, if `sandbox` mode is enabled (planned to become default), the preload has no access to `require`, so preload scripts will need to be bundled!

## To summarise

- If we `init` the renderer SDK in the preload, users with `contextIsolation: false` cannot pass options to `init`
- If we don't `init` the renderer SDK in the preload, users that want errors from preload context will need to use a bundler if `sandbox: true`

**Comment by timfish** [comment]
This is simpler in 3.0.0-beta.2

A preload is no longer required unless you're using Electron < v5 and bundling the main process.

If you would like errors reported from the preload then you'll still need to initialise the renderer portion of the sdk there. No options are required unless you need to configure specific renderer integrations.

`preload.js`
```ts
const { init } = require('@sentry/electron/renderer');
init();
```

If you have context isolation enabled, the renderer is a different context to the preload so you'll need to `init` the sdk again there.

**Comment by its-tim-lee** [comment]
Hey, @timfish 
So did you mean that, if I want to use Sentry in three independent contexts (ie., renderer, preload, main), and if i have enabled both `contextIsolation` and `sandbox`, as long as my Sentry configurations are the same in three contexts, i don't need to configure anything in `init` at the contexts of renderer and preload, since you metioned:

> It's worth noting that as all events are sent from the main process, you do not need to supply the `dsn`, `release` and `environment` in the preload and renderer.

But I guess that I still need to configure the one in renderer, since that's the instruction from the relevant Sentry doc, that I 
need to integrate @sentry/vue by doing something like:
```
  // this is in the renderer context
  Sentry = require("@sentry/electron/renderer")
  Sentry.init(
    /** my setting */,
    require("@sentry/vue").init
  )
```

Am i right?
Currently, I found that if I provide setting to the `Sentry.init` in the preload, I'll get an error, but the error will disappear if I don't provide any setting.

BTW, 
I'm using @sentry/electron v4.20.0, @sentry/vue v7.105.0, and electron v29.1.1
